test/options.test.js

```diff
@@ -141,3 +141,34 @@ exports.callingValueOnSomeReturnsValue = function(test) {
     test.deepEqual(4, options.some(4).value());
     test.done();
 };
+
+exports.filteringNoneReturnsNone = function(test) {
+    function equals3(x) {
+        return x === 3;
+    }
+
+    test.deepEqual(options.none.filter(equals3), options.none);
+    test.done();
+};
+
+exports.filteringSomeReturnsIdentityIfPredicateIsTrue = function(test) {
```

Comment:
This test name doesn't seem quite right to me -- filtering when the predicate is true doesn't return the identity, it's equivalent to applying the identity. I think it'd be clearer to write the test name (including the other tests for filter) similarly to the docs e.g.

```javascript
exports["given predicate(value) is true then some(value).filter(predicate) is some(value)"] = ...
```

What do you think?